presto-main/src/main/java/com/facebook/presto/testing/InterfaceTestUtils.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.testing;
+
+import java.lang.reflect.Method;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public final class InterfaceTestUtils
+{
+    private InterfaceTestUtils() {}
+
+    // Note: this should be in sync with com.facebook.presto.plugin.base.testing.InterfaceTestUtils#checkEverythingImplemented
+    public static <I> void checkEverythingImplemented(Class<I> interfaceClass, Class<? extends I> implementationClass)
+            throws ReflectiveOperationException
+    {
+        checkArgument(interfaceClass.isAssignableFrom(implementationClass));
+        for (Method interfaceMethod : interfaceClass.getMethods()) {
+            Method implementationMethod = implementationClass.getMethod(interfaceMethod.getName(), interfaceMethod.getParameterTypes());
+            if (interfaceMethod.equals(implementationMethod) && interfaceMethod.getAnnotation(Deprecated.class) == null) {
```

Comment:
This equals check will always be false, since the methods will have different declaring classes. Example:
```java
Method a = CharSequence.class.getMethod("length");
Method b = String.class.getMethod("length");
System.out.println(a.getDeclaringClass());
System.out.println(b.getDeclaringClass());
System.out.println(a.equals(b));
```
```
interface java.lang.CharSequence
class java.lang.String
false
```
I think that `TestClassLoaderSafeWrappers` does what you want to do here.

Reply:
> This equals check will always be false, since the methods will have different declaring classes

The intention is to find interface's default method that is not implemented / overridden.
The code is moved here from `TestPartitionsAwareAccessControl`. `TestPartitionsAwareAccessControl` throws the following if i remove one of the implementation methods from `PartitionsAwareAccessControl`:
> ``` java.lang.AssertionError: Method should be overridden in class com.facebook.presto.hive.security.PartitionsAwareAccessControl: public default void com.facebook.presto.spi.connector.ConnectorAccessControl.checkCanCreateSchema(com.facebook.presto.spi.connector.ConnectorTransactionHandle,com.facebook.presto.spi.security.Identity,java.lang.String) ```

> I think that TestClassLoaderSafeWrappers does what you want to do here.

indeed! I didn't about this class, that's why i wrote `TestPartitionsAwareAccessControl` differently. No need for this difference though, we can reuse `TestClassLoaderSafeWrappers`'s logic